libs/types/src/cdf_cast_vote_records.ts

```diff
@@ -0,0 +1,2032 @@
+   */
+  readonly BallotAuditId?: string;
+
+  /**
+   * An image of the ballot sheet created by the scanning device.
+   */
+  readonly BallotImage?: readonly ImageData[];
+
+  /**
+   * A unique identifier for the ballot (or sheet of a multi-sheet ballot) that this CVR represents, used if ballots are pre-marked with unique identifiers.  If provided, this number would be the same on all CVRs that represent individual sheets from the same multi-sheet ballot.  This identifier is not the same as one that may be impressed on the corresponding ballot as it is scanned or otherwise associated with the corresponding ballot; see the BallotAuditId attribute.
+   */
+  readonly BallotPrePrintedId?: string;
+
+  /**
+   * A unique number for the ballot (or sheet of a multi-sheet ballot) that this CVR represents, used if ballots are pre-marked with unique numbers.  If provided, this number would be the same on all CVRs that represent individual sheets from the same multi-sheet ballot.  This number is not the same as one that may be impressed on the corresponding ballot as it is scanned or otherwise associated with the corresponding ballot; see the BallotAuditId attribute.
+   */
+  readonly BallotSheetId?: string;
+
+  /**
+   * An identifier of the ballot style associated with the corresponding ballot.
+   */
+  readonly BallotStyleId?: string;
+
+  /**
+   * Identifies the smallest unit of geography associated with the corresponding ballot, typically a precinct or split-precinct.
+   */
+  readonly BallotStyleUnitId?: string;
+
+  /**
+   * The identifier for the batch that includes this CVR.
+   */
+  readonly BatchId?: string;
+
+  /**
+   * The sequence number of the corresponding paper ballot within a batch.
+   */
+  readonly BatchSequenceId?: integer;
+
+  /**
+   * Identifies the repeatable portion of the CVR that links to contest selections and related information.
+   */
+  readonly CVRSnapshot: readonly CVRSnapshot[];
+
+  /**
+   * Identifies the device that created the CVR.
+   */
+  readonly CreatingDeviceId?: string;
+
+  /**
+   * Identifies the snapshot that is currently tabulatable.
+   */
+  readonly CurrentSnapshotId: string;
+
+  /**
+   * Used to identify an election with which the CVR is associated.
+   */
+  readonly ElectionId: string;
+
+  /**
+   * Identifies the party associated with a CVR, typically for partisan primaries.
+   */
+  readonly PartyIds?: readonly string[];
+
+  /**
+   * The sequence number for this CVR. This represents the ordinal number that this CVR was processed by the tabulating device.
+   */
+  readonly UniqueId?: string;
+
+  /**
+   * Indicates whether a ballot is absentee, precinct, or provisional
+   */
+  readonly vxBallotType: 'absentee' | 'provisional' | 'standard';
+
+  /**
+   * @deprecated Use to include the layout data of the ballot as scanned.
+   * Deprecated as we will move to including these as separate files.
+   */
+  readonly vxLayouts?: SheetOf<Optional<BallotPageLayout>>;
```

Comment:
@jonahkagan I don't want to add this complex type to the JSON schema, so I think I will leave things as _not_ JSON schema based for now and then flip it to the way you're doing it after I move layouts out of this file.